app/utils/EmacUrlBuilder.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import javax.inject.Inject
+
+import config.FrontendAppConfig
+import models.{VatDecEnrolment, VatEnrolment}
+import play.api.mvc.Request
+import uk.gov.hmrc.domain.Vrn
+
+class EmacUrlBuilder@Inject()(appConfig: FrontendAppConfig) {
+
+  def getRequestAccessUrl(enrolmentKey: String)(vatEnrolment: Option[VatEnrolment])(implicit request: Request[_]): String = {
```

Comment:
Everywhere this method is called the `vatEnrolment` is wrapped in a `Some`. It would be better to not make this optional and only wrap it in a `Some` when it is needed